src/components/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,46 @@
+import React = require('react');
+
+interface CheckBoxProps {
+    text?: string;
+    value?: boolean;
+    boxIcon?: string;
+    tickIcon?: string;
+    onChange?: (value: boolean) => any;
+
+}
+
+export const CheckBox: React.SFC<CheckBoxProps> = (props) => {
+
+    const handleClick = (event: React.SyntheticEvent<HTMLElement>): void => {
```

Comment:
you create that function each time it component receiving new props. 

you can easily remove function above
```tsx
function onCheckBoxClick(value: boolean, handler: function): void {
  if (handler) handler(value)
}
//..
onClick={(event) => onCheckBoxClick(!props.value, props.onChange)}>
```